### PR TITLE
fix: DpButton as Link and Input spacing 

### PR DESCRIPTION
**Pull request by salisdemos**
- Fix the determination for defining if dpButton is a button or a link
- add hover-styles to make the button-styled Link looks good
- Adjust Input spacing which for some reason doesn't fit anymore
- a little type scripting

### CODE REVIEW

File: src/components/DpButton/DpButton.vue [hunk 1]
```diff
@@ -263,7 +263,8 @@ const allColorClasses = {
       active:bg-interactive-subtle-active active:text-interactive-active `,
     // solid: classes that only apply to "solid" button color variant.
     solid: `
-      bg-interactive text-on-dark
+      bg-interactive
+      text-on-dark hover:text-on-dark hover:no-underline
       hover:bg-interactive-hover
```
Comment (spiess-demos) [inline src/components/DpButton/DpButton.vue:268]: my idea was to have one line for each state: default state, hover, focus, and so on.

```suggestion
      bg-interactive text-on-dark
      hover:bg-interactive-hover hover:text-on-dark hover:no-underline
```

File: src/components/DpInput/DpInput.vue [hunk 2]
```diff
@@ -41,185 +42,81 @@
 import { computed, ref, toRefs, watch } from 'vue'
 import { exactlengthHint, maxlengthHint, minlengthHint, prefixClass } from '~/utils'
 import DpLabel from '~/components/DpLabel'
-import { length } from '~/shared'
 
-const props = defineProps({
-  /**
```
Comment (spiess-demos) [inline src/components/DpInput/DpInput.vue:47]: Here and below: why did you remove prop documentation?

File: src/components/DpInput/DpInput.vue [hunk 3]
```diff
@@ -41,185 +42,81 @@
...
+    hint: '',
+    text: '',
+    tooltip: ''
+  },
+  maxlength = null,
+  minlength = null,
+  size = null,
+  value = '',
+  width = 'w-full'
+} = toRefs(props)
```
Comment (spiess-demos) [inline src/components/DpInput/DpInput.vue:119]: what is achieved with the toRefs here? i thought props are reactive by default?

File: src/components/DpLabel/DpLabel.vue [hunk 4]
```diff
@@ -23,7 +23,7 @@
 
 <script setup lang="ts">
 import { computed } from 'vue'
-import { CleanHtml as vCleanhtml } from '~/directives' //NOSONAR see https://sonarsource.atlassian.net/browse/JS-97
```
Comment (spiess-demos) [inline src/components/DpLabel/DpLabel.vue:26]: Why did you remove that comment?

File: src/components/DpInput/DpInput.vue [hunk 5]
```diff
@@ -7,7 +7,8 @@
         for: id,
         hint: labelHint,
         required: required
-      }" />
+      }"
+      class="mb-0.5" />
```
Comment (spiess-demos) [inline src/components/DpInput/DpInput.vue:11]: why do you add space here? did you remove it somewhere else?

**Review by spiess-demos — CHANGES_REQUESTED** [review]

File: src/components/DpLabel/DpLabel.vue (hunk 4 above)
Comment (salisdemos) [reply src/components/DpLabel/DpLabel.vue:26]: because they fixed the issue and the linter should not complain anymore 
https://sonarsource.atlassian.net/browse/JS-97

File: src/components/DpInput/DpInput.vue (hunk 3 above)
Comment (salisdemos) [reply src/components/DpInput/DpInput.vue:149]: actually I didn't do it. It was like that before. I don't know why its like that. Git doesn't want to tell me why it was done like that.

**Review by spiess-demos — APPROVED** [review]
lgtm (code review only)
